### PR TITLE
Add `CloudSQLCloneInstanceOperator`

### PR DESCRIPTION
**Pull request by FloMom**
Created CloudSQLCloneInstanceOperator to provide a simple way of cloning a cloudSQL instance from a DAG. 

It is making use of google's instance.clone method as described here: https://cloud.google.com/sql/docs/mysql/admin-api/rest/v1/instances/clone

### CODE REVIEW

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
tests/static checks  are failing

File: tests/always/test_project_structure.py [hunk 1]
```diff
@@ -289,6 +289,7 @@ class TestGoogleProviderProjectStructure(ExampleCoverageTest, AssetsCoverageTest
         "airflow.providers.google.cloud.operators.vertex_ai.endpoint_service.UpdateEndpointOperator",
         "airflow.providers.google.cloud.operators.vertex_ai.batch_prediction_job."
         "GetBatchPredictionJobOperator",
+        "airflow.providers.google.cloud.operators.cloud_sql.CloudSQLCloneInstanceOperator",
```
Comment (eladkal) [inline tests/always/test_project_structure.py]: What is the motivation to add new operator without example?
Is there a difficulty in creating the example dag?
Out goal is to reduce this list not to make it bigger.

**Review by eladkal — CHANGES_REQUESTED** [review]

File: tests/always/test_project_structure.py (hunk 1 above)
Comment (FloMom) [reply tests/always/test_project_structure.py]: Good point. Thank you for the hint. I updated the associated example DAG to demonstrate usage of the CloudSQLCloneInstanceOperator.

Comment (eladkal) [reply tests/always/test_project_structure.py]: Great. We are almost there. The example was created but it is not referenced in the docs so users can't find it.
See our docs https://airflow.apache.org/docs/apache-airflow-providers-google/stable/operators/cloud/cloud_sql.html

You need to add a short description in the docs and refer to the example you wrote.
The relevant file to edit is:
https://github.com/apache/airflow/blob/main/docs/apache-airflow-providers-google/operators/cloud/cloud_sql.rst

Comment (FloMom) [reply tests/always/test_project_structure.py]: all right. Thank you for pointing me to the right files. I updated the PR accordingly.

File: docs/apache-airflow-providers-google/operators/cloud/cloud_sql.rst [hunk 2]
```diff
@@ -473,6 +473,47 @@ More information
...
+    :language: python
+    :dedent: 4
+    :start-after: [START gcp_sql_clone_template_fields]
+    :end-before: [END gcp_sql_clone_template_fields]
+
+More information
+""""""""""""""""
+
+See Google Cloud SQL API documentation to `clone an instance
+<https://cloud.google.com/sql/docs/mysql/admin-api/v1beta4/databases/clone>`_.
```
Comment (eladkal) [inline docs/apache-airflow-providers-google/operators/cloud/cloud_sql.rst]: This link leads to 404
Screenshot 2023-02-28 at 23 16 37

**Comment by eladkal** [comment]
@FloMom static checks are failing :(

**Review by eladkal — APPROVED** [review]

**Pull request merged**